### PR TITLE
[Reviewer: Rob] Add 3GPP-Cx interface to HSS

### PR DESCRIPTION
**Pull request by mirw**
Rob,

Please can you review this change to add a "3GPP-Cx" interface from Homestead to an HSS, along with an HSS proxy charm to test it (and possibly deploy with)?

I've lived-tested on an OpenStack Juju rig, adding and deleting relations, and setting different configuration on the HSS proxy charm and confirming that it is propagated.

Thanks,

Matt

### CODE REVIEW

File: charms/trusty/clearwater-homestead/lib/config_script [hunk 1]
```diff
@@ -23,8 +26,18 @@ local_ip=$(unit-get private-address)
 public_ip=$(unit-get public-address)
 public_hostname=homestead-$(cut -d/ -f2 <<< $JUJU_UNIT_NAME).$(config-get zone)
 node_idx=$(($(cut -d/ -f2 <<< $JUJU_UNIT_NAME) + 1))
+[ "$relation_name" != "hss" ] || [ "$(relation-get public-address)" = "" ] || hss_hostname=$(relation-get public-address)
```
Comment (rkday-pro) [inline charms/trusty/clearwater-homestead/lib/config_script:29]: How does the public-address on the proxy charm get set to the IP of the external HSS?

File: charms/trusty/hss-proxy/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Overview
+
+This is a proxy [Juju charm](https://jujucharms.com/about) for the 3GPP-Cx interface to the HSS.  It does not provide HSS function.
```
Comment (rkday-pro) [inline charms/trusty/hss-proxy/README.md]: Worth a link or fuller explanation of what a proxy charm is? (Googling it isn't very helpful.)

**Comment by rkday-pro** [comment]
Looks good, two comments.

File: charms/trusty/clearwater-homestead/lib/config_script (hunk 1 above)
Comment (mirw) [inline charms/trusty/clearwater-homestead/lib/config_script:29]: I've fixed this up in the proxy charm.

**Pull request merged**